sympy/solvers/solveset.py

```diff
@@ -192,6 +191,11 @@ def inv(trig):
                 invs += Union(*[imageset(Lambda(n, L(g)), S.Integers) for g in g_ys])
             return _invert_real(f.args[0], invs, symbol)
 
+    if (isinstance(a,log) for a in f.args):
+        f_tmp = logcombine(f , force = True)
+        if isinstance(f_tmp,log):
+            invs = E**(list(g_ys)[0])
+            return _invert_real(simplify(f_tmp.args[0]), FiniteSet(invs), symbol)
```

Comment:
it's better to leave simplification to the user. See the docstring of simplify (I believe) for reasons why.